src/tests/ftest/cart/util/cart_utils.py

```diff
@@ -418,6 +418,52 @@ def build_cmd(self, env, host, **kwargs):
 
         return tst_cmd
 
+    def log_check_valgrind_memcheck(self):
+        """Check valgrind memcheck log files for errors."""
+
+        memcheck_errors = 0
+
+        # Write memcheck result file(s) to $HOME or DAOS_TEST_SHARED_DIR.
+        daos_test_shared_dir = os.getenv('DAOS_TEST_SHARED_DIR',
+                                         os.getenv('HOME'))
+
+        self.log.info("Parsing log path %s", daos_test_shared_dir)
+        if not os.path.exists(daos_test_shared_dir):
+            self.log.info("Path does not exist")
+            return 1
+
+        xml_filename_fmt = r"^valgrind\.\S+\.memcheck$"
+        memcheck_files = list(filter(lambda x: re.match(xml_filename_fmt, x),
+                                os.listdir(daos_test_shared_dir)))
+
+        for filename in memcheck_files:
+            log_file = os.path.join(daos_test_shared_dir, filename)
+
+            with open(log_file, 'r') as file1:
+                lines = file1.readlines()
+
+            for line in lines:
+                if line.find('<error>') != -1:
+                    memcheck_errors += 1
+
+        if memcheck_errors > 0:
+            self.log.info(
+                "Found %s <error> element(s) in the " +
+                "memcheck XML log file(s): [%s]",
+                str(memcheck_errors),
+                ", ".join(memcheck_files))
+
+        saved_cwd = os.getcwd()
+        os.chdir(daos_test_shared_dir)
+
+        # Rename the file so it's not checked again for memcheck errors
+        for filename in memcheck_files:
+            os.rename(filename, filename + "-checked")
+
+        os.chdir(saved_cwd)
```

Comment:
You shouldn't be using chdir here
```suggestion
        # Rename the file so it's not checked again for memcheck errors
        for filename in memcheck_files:
            log_file = os.path.join(daos_test_shared_dir, filename)
            os.rename(log_file, log_file + "-checked")
```